packages/orchestrator/lib/clients/validate.ts

```diff
@@ -1,17 +1,64 @@
 import { taskStates } from '@nangohq/scheduler';
 import type { Task } from '@nangohq/scheduler';
-import { TaskAction, TaskWebhook, TaskPostConnection } from './types.js';
+import type { OrchestratorTask } from './types.js';
+import { TaskAction, TaskWebhook, TaskPostConnection, TaskSync } from './types.js';
 import { z } from 'zod';
-import { actionArgsSchema, webhookArgsSchema, postConnectionArgsSchema } from '../routes/v1/postSchedule.js';
 import { Err, Ok } from '@nangohq/utils';
 import type { Result } from '@nangohq/utils';
+import { jsonSchema } from '../utils/validation.js';
+
+export const commonSchemaArgsFields = {
+    connection: z.object({
+        id: z.number().positive(),
+        connection_id: z.string().min(1),
+        provider_config_key: z.string().min(1),
+        environment_id: z.number().positive()
+    })
+};
+
+export const syncArgsSchema = z.object({
+    type: z.literal('sync'),
+    syncId: z.string().min(1),
+    syncName: z.string().min(1),
+    syncJobId: z.number().int().positive(),
+    debug: z.boolean(),
+    ...commonSchemaArgsFields
+});
+
+export const actionArgsSchema = z.object({
+    type: z.literal('action'),
+    actionName: z.string().min(1),
+    activityLogId: z.number().positive(),
+    input: jsonSchema,
+    ...commonSchemaArgsFields
+});
+export const webhookArgsSchema = z.object({
+    type: z.literal('webhook'),
+    webhookName: z.string().min(1),
+    parentSyncName: z.string().min(1),
+    activityLogId: z.number().positive(),
+    input: jsonSchema,
+    ...commonSchemaArgsFields
+});
+export const postConnectionArgsSchema = z.object({
+    type: z.literal('post-connection-script'),
+    postConnectionName: z.string().min(1),
+    fileLocation: z.string().min(1),
+    activityLogId: z.number().positive(),
+    input: jsonSchema,
+    ...commonSchemaArgsFields
+});
```

Comment:
adding syncArgsSchema and moving all the zod schema in `validate.ts` where they belong